contracts/Vault2.sol

```diff
@@ -1,6 +1,41 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
-contract Vault2 {
-    
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract Vault2 is ERC20 {
+    event Minted(address by, uint256 amount);
+    event Burned(address by, uint256 amount);
+
+    constructor() ERC20("VAULT", "VAULT") {}
+
+    modifier noneZeroAmount(uint256 _amount) {
+        require(_amount > 0, "Amount should be greater than 0");
+        _;
+    }
+
+    function mint(uint256 _amount) external payable noneZeroAmount(_amount) {
+        require(
+            _amount == msg.value,
+            "Amount doesn't match the value of ETH sent"
+        );
+
+        _mint(msg.sender, _amount);
+
+        emit Minted(msg.sender, _amount);
```

Comment:
_mint automatically emits an event indicating a mint, you can skip this part - https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L267